### ISSUE TITLE
Anchors and targets

### ISSUE THREAD

**Issue by commorancy** [WordPress/gutenberg#10101]
Since Gutenberg now officially supports on-page anchors per block, this bug has become more relevant. When you set a named anchor point and attempt to reference it, when clicking the live anchor link on the page, the anchor lands at the very top of the page under the WP header and is hidden. If you're not logged in to WordPress, this header is maybe 3 pixels. This means clicking the anchor might look somewhat okay. However, when logged into WP and the thick WP header is present, the anchor point still lands at the page top , but is now under the WP banner and is obscured.

It would be preferable if anchors could become WP header bar aware and move the page so that the anchor always lands a few pixels below the WP header bar so that it always remains visible. As it is now, I always have to move the anchor into the previous paragraph to account for this problem. With blocks, this issue will be more difficult to workaround... particularly if you move blocks around and the anchor was placed in a formerly, but no longer adjacent block to workaround the header bar problem.

**Comment by anevins12** [comment]
Twenty Seventeen binds an event to the inline anchor links and offsets them for the header. Maybe we can add something like that.

**Comment by afercia** [comment]
There are also pure-CSS solutions for this issue, using the `:target` selector. See for example what happens following this link: https://make.wordpress.org/core/handbook/best-practices/commit-messages/#guidelines

**Comment by afercia** [comment]
@wpericam I see in your GIFs the HTML anchor value `Why-use` starts with an uppercase `W` while the URL fragment used in the link has a lowercase one. That makes a difference.

**Comment by wpericam** [comment]
@afercia Thanks for pointing that out - I see what you mean. That doesn't feel intuitive at all, as URLs don't tend to be case sensitive. However, I see that W3.org notes 

> Anchor names must be unique within a document. Anchor names that differ only in case may not appear in the same document and that comparisons between fragment identifiers and anchor names must be done by exact (case-sensitive) match.

TIL today! Thanks for pointing this out!

**Comment by afercia** [comment]
Discussed this issue during latest accessibility bur-scrub on Slack and agreed it should be moved to core Trac. WordPress should take care of target links when the admin bar is displayed. Worth noting Themes that use a sticky toolbar should take care of this in their own CSS.

I'm going to create a Trac ticket and reference it here.

**Comment by afercia** [comment]
Moved to https://core.trac.wordpress.org/ticket/46371
